sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md

```diff
@@ -0,0 +1,413 @@
+
+```C# Snippet:EventHubs_Sample11_MockingConsumerClient
+// Create a mock of the EventHubConsumerClient
+Mock<EventHubConsumerClient> mockConsumer = new Mock<EventHubConsumerClient>();
+
+var receivedEvents = new List<EventData>();
+var cancellationTokenSource = new CancellationTokenSource();
+
+// Create a mock of LastEnqueuedEventProperties using the model factory, these can
+// be set depending on the scenario that is important for the application
+var lastEnqueueEventProperties = EventHubsModelFactory.LastEnqueuedEventProperties(
+    lastSequenceNumber : default,
+    lastOffset : default,
+    lastEnqueuedTime : default,
+    lastReceivedTime : default);
+
+// Create a mock of PartitionContext using the model factory
+var partitionContext = EventHubsModelFactory.PartitionContext(
+    partitionId : "0",
+    lastEnqueuedEventProperties : lastEnqueueEventProperties);
+
+// Mock an EventData instance, different arguments can simulate different potential
+// outputs from the broker
+var eventData = EventHubsModelFactory.EventData(
+    eventBody : new BinaryData("Sample-Event"),
+    systemProperties : default,
+    partitionKey : default,
+    sequenceNumber : default,
+    offset : default,
+    enqueuedTime : default);
+
+// Create a mock of a partition event
+var samplePartitionEvent = new PartitionEvent(partitionContext, eventData);
+var partitionEventList = new List<PartitionEvent>(new PartitionEvent[] { samplePartitionEvent });
+
+// Define a simple local method that returns an IAsyncEnumerable to use as the return for
+// ReadEventsAsync below
+async IAsyncEnumerable<PartitionEvent> mockReturn(PartitionEvent samplePartitionEvent)
+{
+    await Task.CompletedTask;
+    yield return samplePartitionEvent;
+}
+
+// Use this PartitionEvent to mock a return from the consumer, because ReadEvents returns an IAsyncEnumerable a separate
+// method is needed to properly set up this method
+mockConsumer.Setup(
+    c => c.ReadEventsAsync(
+    It.IsAny<CancellationToken>()))
+    .Returns(mockReturn(samplePartitionEvent));
+
+var consumer = mockConsumer.Object;
+```
+
+## Mocking `PartitionReceiver`
+
+The sample below illustrates how to mock a `PartitionReceiver`, and set up its `ReceiveBatchAsync` method to return an empty batch. Mocked partition receivers can be set up for more complex scenarios using `SetUp(...).CallBack(...)` as well.
+
+```C# Snippet:EventHubs_Sample11_PartitionReceiverMock
+// Create a mock of the PartitionReceiver
+var mockReceiver = new Mock<PartitionReceiver>();
+
+// This sets the value returned by the EventDataBatch when accessing the Size property
+// It does not impact TryAdd on the mocked batch
+var batchSizeInBytes = 500;
+
+// Events added to the batch will be added here, but altering the events in this list will not change the
+// events in the batch, since they are stored inside the batch as well
+var backingList = new List<EventData>();
+
+var maximumEventCount = 10;
```

Comment:
This gets used way too far away from the declaration.  I don't think we need the separate variable here, since you've chosen to use named parameters when passing them (which is a great idea)